### PR TITLE
[flyteagent] Remove redundant code in Agent

### PR DESCRIPTION
**Pull request by Future-Outlier**
## Tracking issue
https://github.com/flyteorg/flyte/issues/3936

## Why are the changes needed?
Comment in the description below.

## How was this patch tested?
single binary.

### Screenshots
image

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

File: flyteplugins/go/tasks/plugins/webapi/agent/client.go [hunk 1]
```diff
@@ -63,13 +63,8 @@ func getGrpcConnection(ctx context.Context, agent *Deployment) (*grpc.ClientConn
 	if err != nil {
 		return nil, err
 	}
+
 	defer func() {
-		if err != nil {
-			if cerr := conn.Close(); cerr != nil {
```
Comment (Future-Outlier) [inline flyteplugins/go/tasks/plugins/webapi/agent/client.go:68]: Here will never be triggered

File: flyteplugins/go/tasks/plugins/webapi/agent/plugin.go [hunk 2]
```diff
@@ -175,11 +176,7 @@ func (p Plugin) ExecuteTaskSync(
...
 
 	if err := stream.CloseSend(); err != nil {
-		return nil, nil, err
-	}
-
-	if err != nil {
-		logger.Errorf(ctx, "Failed to write output with err %s", err.Error())
+		logger.Errorf(ctx, "Failed to close stream with err %s", err.Error())
 		return nil, nil, err
 	}
```
Comment (Future-Outlier) [inline flyteplugins/go/tasks/plugins/webapi/agent/plugin.go:181]: Here will never be triggered, too.

File: flyteplugins/go/tasks/plugins/webapi/agent/client.go [hunk 3]
```diff
@@ -63,13 +63,8 @@ func getGrpcConnection(ctx context.Context, agent *Deployment) (*grpc.ClientConn
 	if err != nil {
 		return nil, err
 	}
+
 	defer func() {
```
Comment (pingsutw) [inline flyteplugins/go/tasks/plugins/webapi/agent/client.go:67]: we can remove defer as well

Comment (Future-Outlier) [reply flyteplugins/go/tasks/plugins/webapi/agent/client.go:67]: Yes you are right, nice catch!

**Review by pingsutw — APPROVED** [review]

**Pull request merged**